packages/x-date-pickers/src/internals/hooks/useField/useFieldState.ts

```diff
@@ -207,10 +207,13 @@ export const useFieldState = <
   };
 
   const clearValue = () =>
-    publishValue({
-      value: valueManager.emptyValue,
-      referenceValue: state.referenceValue,
-    });
+    publishValue(
+      {
+        value: valueManager.emptyValue,
+        referenceValue: state.referenceValue,
+      },
+      null,
```

Comment:
The fix is just to avoid passing the sections to `getSectionsFromValue` to make sure that it does not use it to rebuild a value when we want to clear it.